test/clojure/core_test/assoc.cljc

```diff
@@ -0,0 +1,79 @@
+(ns clojure.core-test.assoc
+  (:require [clojure.test :refer [deftest testing is are]]
+            [clojure.core-test.portability #?(:cljs :refer-macros :default :refer) [when-var-exists]]))
+
+(when-var-exists assoc
+  (deftest test-assoc
+
+    (testing "maps"
+      (testing "maps - single value"
+        (are [expected map key val] (= expected (assoc map key val))
+                                    {nil nil} nil nil nil
+                                    {:a 1} nil :a 1
+                                    {:a 1} {} :a 1
+                                    {:a 1 :b 2} {:a 1} :b 2
+                                    {:a 3 :b 2} {:a 1 :b 2} :a 3))
+      (testing "maps - multiple values"
+        (are [expected map kvs] (= expected (apply assoc map kvs))
+                                {:a 1 :b 2} nil [:a 1 :b 2]
+                                {:a 1 :b 2} {} [:a 1 :b 2]
+                                {:a 1 :b 3} {} [:a 1 :b 2 :b 3]
+                                {:a 1 :b 3 :c 5 :d 7} {:a 1 :b 2} [:b 3 :c 5 :d 7]))
+      (testing "maps - sorted type preservation"
+        (is (sorted? (assoc (sorted-map) :a 1 :b 2)))
+        (is (sorted? (assoc (sorted-map :a 1 :b 2) :b 3)))))
+
+    (testing "vectors"
+      (testing "vectors - single value"
+        (are [expected vec index val] (= expected (assoc vec index val))
+                                      [nil] [] 0 nil
+                                      [0] [] 0 0
+                                      [0 3 2] [0 1 2] 1 3
+                                      [0 1 2 3] [0 1 2] 3 3))
+      (testing "vectors - multiple values"
+        ; assoc coll index-1 val-1 index-2 val-2 ...
+        (are [expected vec ivs] (= expected (apply assoc vec ivs))
```

Comment:
Nitpick, `idxs` is more idiomatic in clojure
https://clojuredocs.org/clojure.core/map-indexed#example-542692cdc026201cdc326d25
  https://github.com/clojure/clojure/blob/e6393a4063c42ddc0e0812f04464467764f0fd1e/src/clj/clojure/core.clj#L7492